redis.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentracing
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	tracerLog "github.com/opentracing/opentracing-go/log"
+)
+
+const (
+	operationRedis = "Redis-"
+	logCmdName     = "command"
+	logCmdArgs     = "args"
+	logCmdResult   = "result"
+)
+
+type contextKey int
+
+const (
+	cmdStart contextKey = iota
+)
+
+// jaegerHook is go-redis jaeger hook
+type jaegerHook struct {
+	tracer opentracing.Tracer
+}
+
+// NewJaegerHook return jaegerHook
+func NewJaegerHook(tracer opentracing.Tracer) *jaegerHook {
+	return &jaegerHook{
+		tracer: tracer,
+	}
+}
+
+//BeforeProcess redis before execute action do something
```

Comment:
```suggestion
// BeforeProcess redis before execute action do something
```
the linter seems doesn't work properly, you can use `gofumpt -w -extra .` to fix it.

Reply:
wow，this tool is very good!